test/extensions/filters/http/jwt_authn/authenticator_test.cc

```diff
@@ -136,6 +136,43 @@ TEST_F(AuthenticatorTest, TestCompletePaddingInJwtPayload) {
   EXPECT_EQ(headers.get_("sec-istio-auth-userinfo"), ExpectedPayloadValueWithPadding);
 }
 
+// This test verifies whether the claim is successfully added to header or not
+TEST_F(AuthenticatorTest, TestClaimToHeader) {
+  createAuthenticator();
+  EXPECT_CALL(*raw_fetcher_, fetch(_, _))
+      .WillOnce(Invoke([this](Tracing::Span&, JwksFetcher::JwksReceiver& receiver) {
+        receiver.onJwksSuccess(std::move(jwks_));
+      }));
+
+  Http::TestRequestHeaderMapImpl headers{
+      {"Authorization", "Bearer " + std::string(NestedGoodToken)}};
+
+  expectVerifyStatus(Status::Ok, headers);
+
+  EXPECT_EQ(headers.get_("x-jwt-claim-sub"), "test@example.com");
+  EXPECT_EQ(headers.get_("x-jwt-claim-nested"), "value1");
+  EXPECT_EQ(headers.get_("x-jwt-bool-claim"), "true");
+  EXPECT_EQ(headers.get_("x-jwt-int-claim"), "9999");
+}
+
```

Comment:
Please add a test to validate replace vs. append semantics.

Reply:
These kind of tests are already there - https://github.com/envoyproxy/envoy/pull/23444/files#diff-302a5e5ea85804ebdb3e270f87987fab8dfe876ae1cb85866dedac4833468d52R77